Add scroll to top button

theskumar opened the pull request:
Adds a scroll to top button to all the pages, it shows only if the user scroll past a decent screen size. It's put on the center top so that it's easily accessible. 

**Edit:** Show only when the user is scrolling towards the top.

![Screenshot 2023-11-27 at 5  42 39@2x](https://github.com/HyphaApp/hypha/assets/236356/22ae31da-1454-426d-a638-80215a5eb540)


closes #3642

theskumar committed:
Add scroll to top button

theskumar committed:
put it in a block so it can be removed on any page if needed

theskumar committed:
Accessibility

frjo commented:
Works well.

I prefer the back button near the bottom instead the top of the window. But this is likely a personal preference and might depend on how big your screen is.

theskumar commented:
Thanks for approving. I used this pattern from mkdocs-material. While the button on the right bottom is quite common, there is are couple reason for not placing it there. 

1. While you scroll down, your focus in at the bottom, it cutters that space. 
2. We have a sidebar with quite a lot of buttons, the right bottom will look quite busy. It will also overlap with other buttons. 
3. While scrolling up, the user attention is at the top area, and often forgets to pay attention to the bottom right. 
4. Another thing present on the mkdocs-material, is that it doesn't show the button while you are scrolling down, it shows only when you scrolling in upward direction. While is I find it nice, but didn't implement it here so it's more visible to users + slight simple code/performance gain to track which direction user is scrolling.

theskumar committed:
update styling + show only when scrolling up

theskumar commented:
Updated so it's visible while scrolling up, also adjusted a bit of styling and animation. Let me know if you would prefer this over always visible. After implementing I tend to like this version better. Thoughts?

https://github.com/HyphaApp/hypha/pull/3667/commits/ee3c9859c533d96daa7c0d247aee4434d48565cc

wes-otf commented:
@theskumar I'll be checking this out in the next hour or so. Looks like a great addition!

frjo commented:
Latest version on test in a moment.

wes-otf commented:
All looked and worked very good!

wes-otf commented on hypha/core/templates/components/scroll-to-top.html:
```diff
@@ -0,0 +1,14 @@
+        x-on:scroll.window.throttle.50ms="scrollBackTop = (window.pageYOffset < lastScrollTop && window.pageYOffset > window.outerHeight * 0.4) ? true : false; lastScrollTop = window.pageYOffset;"
+        @click="window.scrollTo({top: 0, behavior: 'smooth'})"
+        aria-label="Back to top"
+        class="fixed top-0 right-1/2 px-3 py-2 mt-10 -mr-[64px] text-white bg-light-blue/80 hover:bg-light-blue hover:text-white hover:shadow-lg transition-all focus:outline-none cursor-pointer shadow-lg rounded-2xl">
```
```suggestion
        class="fixed top-0 right-1/2 px-3 py-2 mt-10 -mr-[64px] text-white bg-light-blue/80 z-30 hover:bg-light-blue hover:text-white hover:shadow-lg transition-all focus:outline-none cursor-pointer shadow-lg rounded-2xl">
```
Did user testing for this today. The only issue I found was it would sometimes get tucked behind/cut off by labels on the submission view. A z-index increase seemed to fix that when I was playing with it. Other than that it's ready for live!

![Screenshot 2023-12-01 at 10 20 57 AM](https://github.com/HyphaApp/hypha/assets/145372368/15bf2c50-0003-48e1-b4c1-e3dd0e56a68d)
![Screenshot 2023-12-01 at 10 07 26 AM](https://github.com/HyphaApp/hypha/assets/145372368/1a224908-1fcb-4ef1-b2c0-e203f973754d)

theskumar replied:
Thank you!

theskumar committed:
Update z-index

Thanks @wes-otf

Co-authored-by: Wes Appler <145372368+wes-otf@users.noreply.github.com>
